cosmic-ext-alternative-startup: init at 0-unstable-2024-11-24; cosmic-ext-extra-sessions: init at 0-unstable-2025-04-02

HeitorAugustoLN opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nyabinary commented on pkgs/by-name/co/cosmic-ext-extra-sessions/package.nix:
```diff
@@ -0,0 +1,166 @@
+    description = "Unofficial alternative sessions for COSMIC Desktop";
+    homepage = "https://github.com/drakulix/cosmic-ext-extra-sessions";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ HeitorAugustoLN ];
```
Should be changed to the COSMIC team now

HeitorAugustoLN replied:
Don't think so, since it is an unofficial package

nyabinary commented on pkgs/by-name/co/cosmic-ext-alternative-startup/package.nix:
```diff
@@ -0,0 +1,53 @@
+    homepage = "https://github.com/Drakulix/cosmic-ext-alternative-startup";
+    license = lib.licenses.gpl3Only;
+    mainProgram = "cosmic-ext-alternative-startup";
+    maintainers = with lib.maintainers; [ HeitorAugustoLN ];
```
Same here. (Should this be changed to the COSMIC team now?)

nyabinary reviewed (commented):
Just a few questions here

HeitorAugustoLN committed:
cosmic-ext-alternative-startup: init at 0-unstable-2024-11-24

HeitorAugustoLN committed:
cosmic-ext-extra-sessions: init at 0-unstable-2025-04-02

HeitorAugustoLN committed:
cosmic-ext-sway-daemon: init at 0-unstable-2025-04-02

HeitorAugustoLN committed:
cosmic-ext-extra-sessions.sway: add cosmic-ext-sway-daemon

HeitorAugustoLN committed:
nixos/cosmic: add alternative session support

This commit introduces options to allow using alternative compositors with the COSMIC desktop environment.

The following options have been added:
- `services.desktopManager.cosmic.extraSessions`: A list of alternative sessions to enable (e.g., "niri", "sway").
- `services.desktopManager.cosmic.defaultSession`: A boolean to control the default `cosmic-comp` session, which can now be disabled.

When an extra session is selected, the corresponding NixOS module for the compositor is enabled, ensuring proper integration. The session files from `pkgs.cosmic-ext-extra-sessions` are also installed.
